website/docs/r/dx_connection.html.markdown

```diff
@@ -20,12 +22,41 @@ resource "aws_dx_connection" "hoge" {
 }
 ```
 
+### Request a MACsec-capable connection
+
+```terraform
+resource "aws_dx_connection" "example" {
+  name             = "tf-dx-connection"
+  bandwidth        = "10Gbps"
+  location         = "EqDA2"
+  macsec_requested = true
+}
+```
+
+### Configure encryption mode for MACsec-capable connections
+-> **NOTE:** You can only specify the `encryption_mode` argument once the connection is in an `Available` state.
+
+```terraform
+resource "aws_dx_connection" "example" {
+  name             = "tf-dx-connection"
+  bandwidth        = "10Gbps"
+  location         = "EqDC2"
+  macsec_requested = true
+  encryption_mode  = "must_encrypt"
+}
+```
+
 ## Argument Reference
 
 The following arguments are supported:
 
 * `bandwidth` - (Required) The bandwidth of the connection. Valid values for dedicated connections: 1Gbps, 10Gbps. Valid values for hosted connections: 50Mbps, 100Mbps, 200Mbps, 300Mbps, 400Mbps, 500Mbps, 1Gbps, 2Gbps, 5Gbps, 10Gbps and 100Gbps. Case sensitive.
+* `encryption_mode` - (Optional) The connection MAC Security (MACsec) encryption mode. MAC Security (MACsec) is only available on dedicated connections. Valid values are `no_encrypt`, `should_encrypt`, and `must_encrypt`.
 * `location` - (Required) The AWS Direct Connect location where the connection is located. See [DescribeLocations](https://docs.aws.amazon.com/directconnect/latest/APIReference/API_DescribeLocations.html) for the list of AWS Direct Connect locations. Use `locationCode`.
+* `macsec_requested` - (Optional) Boolean value indicating whether you want the connection to support MAC Security (MACsec). MAC Security (MACsec) is only available on dedicated connections. See [MACsec prerequisites](https://docs.aws.amazon.com/directconnect/latest/UserGuide/direct-connect-mac-sec-getting-started.html#mac-sec-prerequisites) for more information about MAC Security (MACsec) prerequisites. Default value: `false`.
```

Comment:
The API uses a parameter name of `requestMACSec`. Consider using `request_macsec` as the field name for the schema as well.
```suggestion
* `request_macsec` - (Optional) Boolean value indicating whether you want the connection to support MAC Security (MACsec). MAC Security (MACsec) is only available on dedicated connections. See [MACsec prerequisites](https://docs.aws.amazon.com/directconnect/latest/UserGuide/direct-connect-mac-sec-getting-started.html#mac-sec-prerequisites) for more information about MAC Security (MACsec) prerequisites. Default value: `false`.
```